volatility3/framework/plugins/windows/iat.py

```diff
@@ -0,0 +1,150 @@
+# This file is Copyright 2023 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+
+import logging, io, pefile
+from volatility3.framework.symbols import intermed
+from volatility3.framework import renderers, interfaces, exceptions, constants
+from volatility3.framework.configuration import requirements
+from volatility3.plugins.windows import pslist
+from volatility3.framework.renderers import format_hints
+from volatility3.framework.symbols.windows.extensions import pe
+
+vollog = logging.getLogger(__name__)
+
+
+class IAT(interfaces.plugins.PluginInterface):
+    """Extract Import Address Table to list API (functions) used by a program contained in external libraries"""
+
+    _required_framework_version = (2, 4, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.VersionRequirement(
+                name="pslist", component=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.ListRequirement(
+                name="pid",
+                element_type=int,
+                description="Process ID to include (all other processes are excluded)",
+                optional=True,
+            ),
+        ]
+
+    def _generator(self, procs):
+        kernel = self.context.modules[self.config["kernel"]]
```

Comment:
This is only a lookup in a table, but somehow feels excessive, since the `proc` already knows where the kernel is?  Is `proc.Peb` not already a point to a `_PEB` structure?  If not, then maybe a simple cast to a "pointer to `_PEB`" would be enough (depends how simple that is, it might actually be way more of a pig)?  I'd just be interested to know if that's a thing we should streamline more?  As I say, since it's a lookup happy to keep it this way, it just feels like a disturbance in  the flow somehow...  5;P